### PR TITLE
Regression: Apps-Engine startup process was faulty

### PR DESCRIPTION
**Pull request by tapiarafael**
## Proposed changes (including videos or screenshots)

* App loading routine was forcing apps to be enabled, even if they were previously disabled
* Isolated Apps startup routine for more granular control

## Further comments
Depends on Apps.engine PR - [#594](https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/594)
[AE-109](https://rocketchat.atlassian.net/browse/AE-109)

[AE-109]: https://rocketchat.atlassian.net/browse/AE-109?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/client/views/room/contextualBar/Apps/AppsWithData.tsx [hunk 1]
```diff
@@ -46,7 +47,7 @@ const useValues = (view: IUIKitSurface): [any, Dispatch<any>] => {
 	}));
 
 	const initializer = useMutableCallback(() => {
-		const filterInputFields = (block: IBlock): boolean => {
+		const filterInputFields = (block: IBlock | Block): boolean => {
```
Comment (ggazzo) [inline apps/meteor/client/views/room/contextualBar/Apps/AppsWithData.tsx:50]: sorry to ask but why this change is important?

**Review by ggazzo — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/room/contextualBar/Apps/AppsWithData.tsx (hunk 1 above)
Comment (d-gubert) [reply apps/meteor/client/views/room/contextualBar/Apps/AppsWithData.tsx:50]: The latest `alpha` of Apps-Engine includes a change where we started accepting `Block`s directly from `@rocket.chat/ui-kit` as a peer dependency. However, this caused the types in these callbacks to be invalid and breaking the CI.

It seems this "new" dependency was added to the Apps-Engine before but it had not been updated in Rocket.Chat yet. Then, when we updated the Apps-Engine to support this regression, the dependency changes came along.

**Review by thassiov — APPROVED** [review]
lgtm

**Review by ggazzo — APPROVED** [review]
